packages/cinterop/src/commonMain/kotlin/io/realm/internal/interop/Callback.kt

```diff
@@ -54,3 +54,7 @@ interface SyncLogCallback {
     // Passes core log levels as shorts to avoid unnecessary jumping between the SDK and JNI
     fun log(logLevel: Short, message: String?)
 }
+
+fun interface CompactOnLaunchCallback {
```

Comment:
Don't know if this was what @clementetb suggested. Might be that you can get away with using the in-language functional types and still avoid the boxing. I used `Function3` in 
https://github.com/realm/realm-kotlin/pull/674/files#diff-41ab9e3b640f36477c5ea606fd84f59d64566e8212b5ec307652c8bcde654c56R49, but it doesn't have long arguments, so don't know if it works. Maybe worth a shot, since we don't have to bother with obfuscation, etc.

Reply:
I did use the inbuilt functional types in the first draft. I don't think they have variants with primitive arguments...or at least I don't how to call them.